creator-node/src/routes/tracks.js

```diff
@@ -37,12 +41,101 @@ const RehydrateIpfsQueue = require('../RehydrateIpfsQueue')
 const { FileProcessingQueue } = require('../FileProcessingQueue')
 const DBManager = require('../dbManager')
 const { generateListenTimestampAndSignature } = require('../apiSigning.js')
+const DiskManager = require('../diskManager')
 
 const readFile = promisify(fs.readFile)
+const tmpTrackArtifactsPath = DiskManager.getTmpTrackUploadArtifactsPath()
+const resumableUploadRoute = path.join(tmpTrackArtifactsPath, '*', '*')
+
+// Structure: <uuid>.<file_extension>. e.g. 1234-1234-1234-1234.mp3
+const getFileName = (req) => {
+  return req.randomFolderName + '/' + req.randomFileName + getFileExtension(req.headers.filename)
+}
+
+const server = new tus.Server()
+server.datastore = new tus.FileStore({
+  path: tmpTrackArtifactsPath, // has to be a path that exists
+  namingFunction: getFileName
+})
+
+/**
+ * The helper method to do the resumable upload and chunking
+ */
+async function handleResumableUpload (req, res, next) {
+  try {
+    // Grab the file details off of the url
+    const urlArr = req.originalUrl.split('/')
+    const fileDir = (urlArr.slice(0, urlArr.length - 1)).join('/')
+    const randomFolderName = (urlArr.slice(urlArr.length - 2))[0]
+    const randomFileName = (urlArr.slice(urlArr.length - 1))[0]
```

Comment:
this is a uniqueFileName not a randomFileName now right? I assume this works because the URL is somehow uniquely identifying the track? if so maybe we could add some comments for the future reader to understand that

Reply:
ah yes will do

Reply:
did we change this?

Reply:
yup! https://github.com/AudiusProject/audius-protocol/pull/1484/files#diff-cea0356d19f998f416dcaca2141fb24156c529095b61daebb71f8e3316071f32R116-R119